8.x/theme-development/helpers.md

```diff
@@ -2,4 +2,2097 @@
+                }
+            }
+        }
+        return $selected_row;
+    }
+}
+```
+
+
+### get_cart()
+-------------------------------------------
+
+
+```php
+if (!function_exists('get_cart')) {
+    /**
+     * @return \Illuminate\Database\Eloquent\Collection|CartRow[]
+     */
+    function get_cart(): \Illuminate\Database\Eloquent\Collection|array
+    {
+        $customer = get_customer_user();
+        if ($customer !== false) {
+            $cart_rows = $customer->cartRows()->with('product')->orderBy('id', 'DESC')->get();
+        } else {
+            $cart_rows = get_local_cart(true);
+        }
+        return $cart_rows;
+    }
+}
+```
+
+
+### get_breadcrumb()
+This function will be deprecated in the next version.
+
+```php
+if (!function_exists('get_breadcrumb')) {
+      /**
+     * @deprecated
+     */
+    function get_breadcrumb(Directory $directory): string
+    {
+        if (!(isset($directory->parentDirectory)))
+            return '<li class="active">' . $directory->title . '</li>';
+        else {
+            return "<li><a href=" . $directory->url_full . ">" . $directory->title . "</a></li>"
+                . get_breadcrumb($directory->parentDirectory);
+        }
+    }
+}
+```
+
+
+### get_minimum_purchase_free_shipment()
+This function specifies how much to buy to make shipping free.
+
+```php
+if (!function_exists('get_minimum_purchase_free_shipment')) {
+    function get_minimum_purchase_free_shipment()
+    {
+        try {
+            return ShipmentCostService::getRecord()->getMinimumPurchaseFreeShipment();
+        } catch (Exception $e) {
+            Log::error('Message : ' . $e->getmessage());
+            return 'not set';
+        }
+    }
+}
+```
+
+
+### product_disable_on_min()
+This function specifies when a product is disabled, when it is zero, or when it reaches a minimum.
+
+```php
+if (!function_exists('product_disable_on_min')) {
+    function product_disable_on_min(): ?string
+    {
+        return Product::shouldDisableOnMin();
+    }
+}
+```
+
+
+### get_state_deactivate_product()
+This function will be deprecated in the next version.
+
+```php
+if (!function_exists('get_state_deactivate_product')) {
+      /**
+     * @deprecated
+     */
+    function get_state_deactivate_product($product)
+    {
+        return $product->inaccessibility_type;
+    }
+}
+```
+
+
+### get_inquiry_call_number()
+This function will be deprecated in the next version.
+
+```php
+if (!function_exists('get_inquiry_call_number')) {
+      /**
+     * @deprecated
+     */
+    function get_inquiry_call_number(): string
+    {
+        try {
+            return Setting::getCMSRecord(CMSSettingKey::INQUIRY_CALL_NUMBER)->value;
+        } catch (Exception $e) {
+            Log::error('Message : ' . $e->getmessage());
+            return 'not set';
+        }
+    }
+}
+```
+
+
+### customer_can_edit_profile()
+This function specifies whether the customer is allowed to edit their profile.
```

Comment:
This function specifies whether the customer is allowed to edit their profile or not.